demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -3219,34 +3219,81 @@ export class DemoMeetingApp
         }
   }
 
-  createPinUnpinListener(tileState: VideoTileState): (event: Event) => void {
+  convertTargetResolutionValue(value: string): TargetDisplaySize {
+    switch (value) {
+      case 'low':
+        return TargetDisplaySize.Low
+      case 'medium':
+        return TargetDisplaySize.Medium
+      case 'high':
+        return TargetDisplaySize.High
+    }
+  }
+
+  createTargetResolutionListener(tileState: VideoTileState, form: HTMLFormElement): (event: Event) => void {
     return (event: Event): void => {
+      if (!(event.target instanceof HTMLInputElement)) {
+        // Ignore the Label event which will have a stale value
+        return
+      }
       const attendeeId = tileState.boundAttendeeId;
-        if (this.roster[attendeeId].pinned ) {
-          (event.target as HTMLButtonElement).innerText = 'Pin';
-          this.roster[attendeeId].pinned = false;
-        } else {
-          (event.target as HTMLButtonElement).innerText = 'Unpin';
-          this.roster[attendeeId].pinned = true;
-        }
-        this.updateDownlinkPreference();
+      const value: string = (form.elements.namedItem('level') as RadioNodeList).value;
+      this.log(`target resolution changed for: ${attendeeId} to ${value}`);
+      if (this.attendeeIdToVideoPreference.has(attendeeId)) {
+        this.attendeeIdToVideoPreference.get(attendeeId).targetSize = this.convertTargetResolutionValue(value);
+      } else {
+        this.attendeeIdToVideoPreference.set(attendeeId, new VideoPreference(attendeeId, DemoMeetingApp.DEFAULT_PRIORITY, this.convertTargetResolutionValue(value)));
       }
+      this.updateDownlinkPreference();
+    }
+  }
+
+  convertPriorityValue(value: string): number {
+    switch (value) {
+      case 'low':
+        return 10
+      case 'medium':
+        return 5
+      case 'high':
+        return 0
```

Comment:
According to our [guide](https://github.com/aws/amazon-chime-sdk-js/pull/1593/files#diff-bf95afb6c6ffb26425700a8ed44789aa169e96cd3e93679a60122e2ca0a2e743R13) 1 is the highest priority so its probably should be 1 here.